app/src/main/java/com/lehtimaeki/askold/landingscreen/LandingScreenFragment.kt

```diff
@@ -1,176 +1,198 @@
-
-            if(dataSet[position].iconSet?.tintForContrast == true){
-                ImageViewCompat.setImageTintList(
-                    viewHolder.iconView,
-                    ColorStateList.valueOf(ColorPalettes.getContrastColor(color))
-                )
+    @Composable
+    fun Item(iconSetWrapper: IconSetWrapper) {
+        if (iconSetWrapper.iconSet == null) {
+            Row(
+                verticalAlignment = Alignment.CenterVertically,
+                horizontalArrangement = Arrangement.Start
+            ) {
+                iconSetWrapper.label?.let { TitleText(it) }
             }
-
-            viewHolder.iconView.setImageResource(
-                dataSet[position].iconSet?.icons?.first()
-                    ?: throw RuntimeException("ended up in wrong viewholder bind")
-            )
-
-            viewHolder.itemView.setOnClickListener {
-                activity?.startActivity(Intent(context, FullscreenActivity::class.java).apply {
-                    putExtras(Bundle().apply {
-                        putInt(ICON_SET_EXTRA_ID, dataSet[position].id)
-                        putBoolean(IS_LARGE_CARD_MODE_ID, binding.bigTileSwitch.isChecked)
-                    })
-                })
+        } else {
+            val color =
+                ColorPalettes.getNextColorFromPalette(iconSetWrapper.iconSet.useLightPalette)
+            Column {
+                Card(
+                    modifier = Modifier
+                        .padding(top = 8.dp, bottom = 4.dp, start = 16.dp, end = 16.dp)
+                        .clip(RoundedCornerShape(16.dp))
+                        .size(140.dp)
+                        .clickable(onClick = { navigateToFullScreenActivity(iconSetWrapper) }),
+                    elevation = 8.dp,
+                    backgroundColor = Color(color)
+                ) {
+                    val text = if (iconSetWrapper.iconSet.isUnlocked) "FREE" else "PAID"
+                    Box {
+                        ItemImage(
+                            iconSetWrapper.iconSet.icons.firstOrNull(),
+                            modifier = Modifier.align(Alignment.Center)
+                        )
+                        ItemTypeText(text = text, modifier = Modifier.align(Alignment.TopEnd))
+                    }
+                }
+                ItemNameText(iconSetWrapper.iconSet.name)
             }
         }
+    }
 
-
-        private fun onBindViewHolder(viewHolder: TitleViewHolder, position: Int) {
-            viewHolder.textView.text = dataSet[position].label
-        }
-
-        override fun getItemCount() = dataSet.size
+    private fun navigateToFullScreenActivity(iconSetWrapper: IconSetWrapper) {
+        activity?.startActivity(
+            Intent(
+                context,
+                FullscreenActivity::class.java
+            ).apply {
+                putExtras(Bundle().apply {
+                    putInt(ICON_SET_EXTRA_ID, iconSetWrapper.iconSet!!.id)
+                })
```

Comment:
You can use the `bundleOf` top-level function for code brevity

```suggestion
                putExtras(bundleOf(ICON_SET_EXTRA_ID, iconSetWrapper.iconSet!!.id))
```

Reply:
Also, avoid the double bang operator (`!!`), preferring `requireNotNull { ... }` or throw a custom exception